### PR TITLE
Add some proposed changes

### PR DESCRIPTION
**Pull request by deathaxe**

### CODE REVIEW

**Comment by deathaxe** [comment]
After some research, I guess the first commit _
[Git Files] Config: Allow single quoted value strings_ seems to be wrong. I was mislead somehow

**Comment by y0ssar1an** [comment]
Sorry I missed this PR until now. I'll review/merge it tonight (PST). If you send me review requests, I'll be more likely to catch your PRs.

File: Git Files/Git Config.sublime-syntax [hunk 1]
```diff
@@ -131,6 +153,20 @@ contexts:
     - match: \b(?:yes|true|on|off|no|false)\b
       scope: constant.language.git.config
 
+  shell-script:
+    - match: (\")(\!)
+      captures:
+        1: string.quoted.double.git.config punctuation.definition.string.begin.git.config
+        2: keyword.control.import.shell.git.config
+      escape: \1$
```
[the remark this answers is not in the document]
Comment (michaelblyons) [reply Git Files/Git Config.sublime-syntax]: I am guessing the `(\")`&hellip;`\1` was initially to support single-quote strings (which, to my chagrin, are not allowed).

File: Git Files/Git Config.sublime-syntax [hunk 2]
```diff
@@ -164,7 +164,7 @@ contexts:
       embed: scope:source.shell
     - match: \!
       scope: keyword.control.import.shell.git.config
-      escape: $
+      escape: (?<!\\)\n
```
Comment (michaelblyons) [inline Git Files/Git Config.sublime-syntax]: This is helpful for multiline aliases, but I think someone might object to using the lookbehind.

File: Git Files/Git Config.sublime-syntax [hunk 3]
```diff
@@ -139,45 +175,56 @@ contexts:
         - meta_scope: string.quoted.double.git.config
         - match: \"
           scope: punctuation.definition.string.end.git.config
-          set: expect-newline
-        - include: trailing-slash
+          set: expect-line-end
+        - include: line-continuation
         - include: escape
-        - include: illegal-newline  # newlines only permitting after trailing \
+        - include: illegal-line-end  # newlines only permitting after trailing \
```
[the remark this answers is not in the document]
Comment (deathaxe) [reply Git Files/Git Config.sublime-syntax]: I am  pretty sure not to be the author of the comment ;-)

According to git blame it was added by you on 2017/09/16

Will fix it.

File: Git Files/Git Config.sublime-syntax (hunk 2 above)
Comment (deathaxe) [reply Git Files/Git Config.sublime-syntax]: This lookbehind was the only way to let the embedded shell scope to highlight up to the last valid character and respect the trailing backslash for proper line continuation.

This lookbehind 

1. does not break sregex compatibility according to "Syntax Tests - Check Compatibility"
2. does not have any effect on performance at all

Better solutions without any drawbacks and less code changes needed ;-) are very welcome.

File: Git Files/Git Config.sublime-syntax (hunk 1 above)
[the remark this answers is not in the document]
Comment (deathaxe) [reply Git Files/Git Config.sublime-syntax]: It doesn't matter whether `\1` or `\"` is used here. I just read from @wbond backrefs are supported, so I tried and kept. Not a big deal to change it to `\"` if you like it better.

File: Git Files/Git Config.sublime-syntax (hunk 3 above)
[the remark this answers is not in the document]
Comment (y0ssar1an) [reply Git Files/Git Config.sublime-syntax]: lol thanks for the fix! 😂 👍

**Pull request merged**